api/app/src/main/kotlin/packit/AppConfig.kt

```diff
@@ -1,21 +1,24 @@
 package packit
 
+import org.springframework.boot.context.properties.EnableConfigurationProperties
 import org.springframework.context.annotation.Bean
 import org.springframework.core.env.Environment
 import org.springframework.core.env.get
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
 import org.springframework.security.crypto.password.PasswordEncoder
 import org.springframework.stereotype.Component
+import packit.config.TaskExecutorProperties
 
 @Component
-class AppConfig(private val enviroment: Environment)
+@EnableConfigurationProperties(TaskExecutorProperties::class)
```

Comment:
I don't think you need `EnableConfigurationProperties`. I've used `ConfigurationProperties` in other places fine without it (see `ServiceLoginConfig`). There's a `@ConfigurationPropertiesScan` annotation on `packit.App` that automatically loads any `ConfigurationProperties` found in the `packit` package.